common/src/TrenchBroomApp.cpp

```diff
@@ -225,32 +246,32 @@ namespace TrenchBroom {
                 std::tie(gameName, mapFormat) = gameFactory.detectGame(path);
                 
                 if (gameName.empty() || mapFormat == Model::MapFormat::Unknown) {
-                    if (!GameDialog::showOpenDocumentDialog(NULL, gameName, mapFormat))
+                    if (!GameDialog::showOpenDocumentDialog(nullptr, gameName, mapFormat))
                         return false;
                 }
 
                 frame = m_frameManager->newFrame();
 
                 Model::GameSPtr game = gameFactory.createGame(gameName, frame->logger());
-                ensure(game.get() != NULL, "game is null");
+                ensure(game.get() != nullptr, "game is null");
 
                 frame->openDocument(game, mapFormat, path);
                 return true;
             } catch (const FileNotFoundException& e) {
                 m_recentDocuments->removePath(IO::Path(path));
-                if (frame != NULL)
+                if (frame != nullptr)
                     frame->Close();
-                ::wxMessageBox(e.what(), "TrenchBroom", wxOK, NULL);
+                ::wxMessageBox(e.what(), "TrenchBroom", wxOK, nullptr);
                 return false;
             } catch (const Exception& e) {
-                if (frame != NULL)
+                if (frame != nullptr)
                     frame->Close();
-                ::wxMessageBox(e.what(), "TrenchBroom", wxOK, NULL);
+                ::wxMessageBox(e.what(), "TrenchBroom", wxOK, nullptr);
                 return false;
             } catch (...) {
```

Comment:
`TrenchBroomApp::openDocument` should catch `RecoverableException` here.
To reproduce: corrupt a compilation profile cfg, then load a map from TB's "welcome dialog". You'll get the standard wxMessageBox instead of the one offering to delete the cfg.